src/Plugin/ContainerAwarePluginManager.php

```diff
@@ -62,10 +63,17 @@ public function createInstance($plugin_id, array $configuration = array()) {
     $plugin_definition += array(
       'arguments' => array(),
     );
+    $plugin_class = static::getPluginClass($plugin_id, $plugin_definition);
+
+    // If the plugin provides a factory method, pass the container to it.
+    if (is_subclass_of($plugin_class, 'Drupal\service_container\Plugin\ContainerAwarePluginManager')) {
```

Comment:
This needs to be the Interface that I outlined, not our own.

We are okay to add that interface, too.